File: app-python/entitlements_app/requests/set_account_override_http_request.py
"""Caller-visible HTTP body for command `SetAccountOverride`.

Request-context fields are supplied by authenticated transport state, not caller JSON.
"""

from pydantic import BaseModel


class SetAccountOverrideHttpRequest(BaseModel):
    allowance: int
    createdAt: str
    expiresAt: str | None = None
    featureKey: str
    overrideId: str
    overrideType: str
    reason: str
    tenantId: str

    @property
    def created_at(self) -> str:
        return self.createdAt

    @property
    def expires_at(self) -> str:
        return self.expiresAt

    @property
    def feature_key(self) -> str:
        return self.featureKey

    @property
    def override_id(self) -> str:
        return self.overrideId

    @property
    def override_type(self) -> str:
        return self.overrideType

    @property
    def tenant_id(self) -> str:
        return self.tenantId

File: app/src/main/java/com/example/spring/SetAccountOverrideHttpRequest.java
package com.example.spring;

/**
 * Caller-visible HTTP body for command SetAccountOverride. Path variables and request-context fields
 * are supplied by transport state, not caller JSON.
 */
public record SetAccountOverrideHttpRequest(
    int allowance,
    @jakarta.validation.constraints.NotBlank String createdAt,
    String expiresAt,
    @jakarta.validation.constraints.NotBlank String featureKey,
    @jakarta.validation.constraints.NotBlank String overrideId,
    @jakarta.validation.constraints.NotBlank String overrideType,
    @jakarta.validation.constraints.NotBlank String reason,
    @jakarta.validation.constraints.NotBlank String tenantId
) {}
